Babylon/groups/api/groups/solution/commands/create.py

```diff
@@ -1,49 +1,117 @@
-        except cosmotech_api.exceptions.UnauthorizedException as _e:
-            logger.error("Unauthorized access to the cosmotech api")
+@pass_solution_api
+@pass_environment
+@argument("solution_file", required=False)
+@require_deployment_key("solution_version", "solution_version")
+@require_deployment_key("solution_repository", "solution_repository")
+@require_deployment_key("organization_id", "organization_id")
+@option(
+    "-e",
+    "--use-working-dir-file",
+    "use_working_dir_file",
+    is_flag=True,
+    type=bool,
+    help="Should the path be relative to the working directory ?",
+)
+@option(
+    "-n",
+    "--name",
+    "solution_name",
+    required=True,
+    help="New solution name",
+)
+@option(
+    "-d",
+    "--description",
+    "solution_description",
+    help="New solution description",
+)
+@option(
+    "-s",
+    "--select",
+    "select",
+    type=bool,
+    help="Select this new Solution as babylon context solution ?",
+    default=True,
+)
+def create(
+    env: Environment,
+    solution_api: SolutionApi,
+    organization_id: str,
+    solution_name: str,
+    solution_version: str,
+    solution_repository: str,
+    solution_file: str,
+    select: bool,
+    solution_description: Optional[str] = None,
+    use_working_dir_file: Optional[bool] = False,
+    dry_run: bool = False,
+):
+    """Send a JSON or YAML file to the API to create a solution."""
+
+    if dry_run:
+        logger.info("DRY RUN - Would call solution_api.create_solution")
+        return
+
+    converted_solution_content = get_api_file(
+        api_file_path=solution_file
+        if solution_file
+        else f"{TEMPLATE_FOLDER_PATH}/working_dir_template/API/Solution.yaml",
+        use_working_dir_file=use_working_dir_file if solution_file else False,
+        logger=logger,
+    )
+    if not converted_solution_content:
+        logger.error("Error : can not get correct solution definition, please check your Solution.YAML file")
+        return
+
+    if not solution_description and "solution_description" not in converted_solution_content:
+        converted_solution_content["description"] = solution_name
+
+    converted_solution_content["name"] = solution_name
+    converted_solution_content["key"] = solution_name.replace(" ", "")
+    converted_solution_content["version"] = solution_version
+    converted_solution_content["repository"] = solution_repository
+
+    try:
+        retrieved_solution = solution_api.create_solution(
+            organization_id=organization_id, solution=converted_solution_content
+        )
+    except UnauthorizedException:
+        logger.error("Unauthorized access to the cosmotech api")
+        return
+    except NotFoundException:
+        logger.error(f"Organization with id {organization_id} does not exists.")
+        return
+
+    if select:
+        env.configuration.set_deploy_var("solution_id", retrieved_solution["id"])
+
```

Comment:
Output the retrieved solution in a file if the output file option has been specified